.github/workflows/ci.yaml

```diff
@@ -75,38 +77,37 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
-      - name: Install Poetry
-        uses: abatilo/actions-poetry@v2.4.0
+      - name: Install uv and set the python version
+        uses: astral-sh/setup-uv@v5
         with:
-          poetry-version: 2.0.1
+          version: 0.7.2
+          python-version: ${{ matrix.python-version }}
 
       - name: Install Airflow & dbt
         run: |
-          poetry env use ${{ matrix.python-version }}
-          poetry add "apache-airflow~=${{ matrix.airflow-version }}.0" \
-                     "dbt-core~=${{ matrix.dbt-version }}.0" \
-                 --python ${{ matrix.python-version }}
-          poetry install -E adapters --with dev
-          poetry run airflow db migrate
-          poetry run airflow connections create-default-connections
+          uv add "apache-airflow~=${{ matrix.airflow-version }}.0" \
+                 "dbt-core~=${{ matrix.dbt-version }}.0"
+          uv sync --all-extras --dev
+          uv run airflow db migrate
+          uv run airflow connections create-default-connections
 
       - name: Linting with ruff
-        run: poetry run ruff check .
+        run: uv run ruff check .
 
       - name: Static type checking with mypy
         # We only run mypy on the latest supported versions of Airflow & dbt,
         # so it is currently impossible to write conditions for that depend on package versions.
         if: matrix.python-version == '3.12' && matrix.airflow-version == '2.10' && matrix.dbt-version == '1.9'
-        run: poetry run mypy .
+        run: uv run mypy .
 
-      - name: Code formatting with black
-        run: poetry run black --check .
+      - name: Code formatting with ruff
+        run: uv run ruff format --check .
```

Comment:
Also getting rid of black. Sometimes we use ruff, sometimes black. This is just consolidating under one tool.